sdk/core/System.ClientModel/src/Options/LoggingOptions.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace System.ClientModel.Options;
+
+/// <summary>
+/// TODO
+/// </summary>
+public class LoggingOptions
+{
+    private bool _frozen;
+
+    private const int DefaultLoggedContentSizeLimit = 4 * 1024;
+    private const bool DefaultIsLoggingEnabled = true;
+    private const bool DefaultIsLoggingContentEnabled = false;
+    private static readonly IList<string> s_defaultLoggedHeaderNames =
```

Comment:
I believe that the way this is currently used (where _loggedHeaderNames is assigned to this static list instance), modifications to any LoggingOptions instance in a process will change other instances, which all point back to this.  That's probably not the desired behavior?  If that's correct, we have a few options.  We could keep a static IReadOnlyList that we copy into un-frozen instances of LoggingOptions.  We could also add a private method that creates a new mutable List instance with the default values already in them.  The latter is possible better from a perf perspective?